account-kit/core/src/actions/disconnect.ts

```diff
@@ -46,17 +46,36 @@ export async function disconnect(config: AlchemyAccountsConfig): Promise<void> {
 // Persistence of Wallet Connect connection state on error.
 const clearWalletConnectStore = () => {
   // Open Wallet Connect Indexed DB
+  let walletConnectDBExists = true;
   const dbOpenRequest = indexedDB.open("WALLET_CONNECT_V2_INDEXED_DB");
 
+  dbOpenRequest.onupgradeneeded = () => {
+    if (dbOpenRequest.result.version === 1) {
+      walletConnectDBExists = false;
+
+      // Remove the Database created from the indexedDB.open() call above.
+      indexedDB.deleteDatabase("WALLET_CONNECT_V2_INDEXED_DB");
```

Comment:
wouldn't this throw if this state means it's not connected?

Reply:
Actually, by this point the DB would actually already have been created by the previous `indexedDB.open()` call made above. It's the behaviour for IDB to create a database if it doesn't exist when we call `indexedDB.open()`.

Reply:
I see, so lemme run this back to see if I understand

`db.open` will create a db if it doesn't exist or open the existing one if it does.

if it doesn't exist and thus gets created, that fires the upgrade event here. We know it was created because version is 1 indexed (not 0 indexed?)

Since this only gets called on disconnect, there's no risk that this gets fired when WC first creates the db